### ISSUE TITLE
Spectre v4 shown as mitigated, even if mitigation is disabled

### ISSUE THREAD

**Issue by sblive** [speed47/spectre-meltdown-checker#210]
Hi!

First: thanks for your great script!

I think there is a bug in checking v4 mitigation: if it is disabled via kernel option "spec_store_bypass_disable=off" it will still show "STATUS: NOT VULNERABLE" in green, at least on a system with SSBD uCode.
(disabling v2 via "spectre_v2=off" will correctly show vulnerable in red for v2, same for PTI/Meltdown)

Regards

**Comment by speed47** [comment]
Thanks for your report,

I don't have any CPU with the proper microcode to test myself: could you run the script again in verbose mode (`-v -v`) when you have set `spec_store_bypass_disable=off` in your kernel command-line?

**Comment by sblive** [comment]
I should have said I think it's using the wrong flag for overall vulnerability status, because it shows "mitigated according to /sys: no", but then "not vulnerable" anyway.

Here's the -v - output:

CVE-2018-3640 [rogue system register read] aka 'Variant 3a'
* CPU microcode mitigates the vulnerability:  YES 
> STATUS:  NOT VULNERABLE  (your CPU microcode mitigates the vulnerability)

CVE-2018-3639 [speculative store bypass] aka 'Variant 4'
* Mitigated according to the /sys interface:  NO  (Vulnerable)
(debug) sys_interface_check: /sys/devices/system/cpu/vulnerabilities/spec_store_bypass=Vulnerable
* Kernel supports speculation store bypass: (debug) found Speculation.Store.Bypass: in /proc/self/status
 YES  (found in /proc/self/status)
> STATUS:  NOT VULNERABLE  (your system provides the necessary tools for software mitigation)

**Comment by sblive** [comment]
It's the same on an Epyc system with SSBD:

CVE-2018-3639 [speculative store bypass] aka 'Variant 4'
* Mitigated according to the /sys interface:  NO  (Vulnerable)
(debug) sys_interface_check: /sys/devices/system/cpu/vulnerabilities/spec_store_bypass=Vulnerable
* Kernel supports speculation store bypass: (debug) found Speculation.Store.Bypass: in /proc/self/status
 YES  (found in /proc/self/status)
> STATUS:  NOT VULNERABLE  (your system provides the necessary tools for software mitigation)

**Comment by andrew-aitchison** [comment]
I have the same problem on a RHEL6 machine (well Scientific Linux 6 really) with both the default kernel and the kernel from elrepo-kernel kernel-lt package.
These two kernels have Speculation_Store_Bypass in the /proc/self/status output, but with (different) strings indicating that there is still a vulnerability:
```
% uname -r
2.6.32-754.6.3.el6.x86_64
% grep -E 'Speculation.?Store.?Bypass:' /proc/self/status
Speculation_Store_Bypass:       thread vulnerable
-------------------------------------------------
% uname -r
4.4.163-1.el6.elrepo.x86_64
% grep -E 'Speculation.?Store.?Bypass:' /proc/self/status
Speculation_Store_Bypass:       vulnerable
```
Basically,  matching Speculation_Store_Bypass in /proc/self/status does not mean that all is well:-(

[proc.self.status.2.6.32-754.6.3.el6.x86_64.txt](https://github.com/speed47/spectre-meltdown-checker/files/2604593/proc.self.status.2.6.32-754.6.3.el6.x86_64.txt)
[github.210.vvoutput.txt](https://github.com/speed47/spectre-meltdown-checker/files/2604598/github.210.vvoutput.txt)

**Comment by speed47** [comment]
Latest commit should fix this issue (b022b27), I tested it on my systems, but can you confirm it works for you too?

**Comment by sblive** [comment]
works fine for me now thanks!
